### PR TITLE
Support options for Explain

### PR DESCRIPTION
**Pull request by AnEmortalKid**
Supports some of the postgres options for Explain based on: https://www.postgresql.org/docs/current/using-explain.html

I made the classes take values just in case feedback from #993 comes back as needing to know what those values are (or if they are needed in the future). Might need some clarification there since there's a piece of doc I found that mentioned the order was [ANALYZE VERBOSE] or something with parenthesis, so depending on the postrgres version this might change.

### CODE REVIEW

**Comment by auntyellow** [comment]
It's better to add `getOptionType` in `ExplainStatement.Option`, or simply add a `Map<OptionType, String>` into `ExplainStatement`.
I have to write too much codes to detect this option, e.g.:

			CCJSqlParser parser = new CCJSqlParser(new StringProvider(sql));
			Statement st = parser.Statement();
			if (st instanceof Select) {
				....
			} else if (st instanceof ExplainStatement) {
				ExplainStatement es = (ExplainStatement) st;
				List<Option> options = es.getOptions();
				for (int i = 0; i < options.size(); i ++) {
					if (options.get(i).formatOption().equals("VERBOSE")) {
						...
                                        }
				}
			}

It can be simplified like:

				ExplainStatement es = (ExplainStatement) st;
				if (st.getOptionMap().containsKey(OptionType.VERBOSE)) {
					...
                                }

**Comment by AnEmortalKid** [comment]
@auntyellow how would you feel about

```
Option verbose = ExplainStatement.getOption(OptionType.VERBOSE);
```

Since they _can_ have a value of like TRUE,FALSE,0,1,NO,YES (hadn't gotten to that parsing if it wasn't needed)

**Comment by auntyellow** [comment]
> ```
> Option verbose = ExplainStatement.getOption(OptionType.VERBOSE);
> ```
> 
That's good.

**Comment by AnEmortalKid** [comment]
Cool i'll work on getting that part updated then :)

File: pom.xml [hunk 1]
```diff
@@ -413,6 +413,29 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
```
Comment (wumpz) [inline pom.xml:416]: Why did you add all those profiles? The main build seems to be fast enough even with all this enabled.

File: src/main/java/net/sf/jsqlparser/statement/ExplainStatement.java [hunk 2]
```diff
@@ -11,9 +11,16 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * An {@code EXPLAIN} statement
+ */
 public class ExplainStatement implements Statement {
 
     private Select select;
+    private List<Option> options;
```
Comment (wumpz) [inline src/main/java/net/sf/jsqlparser/statement/ExplainStatement.java:23]: I don't know exactly if an option can be here multiple times. So if then a Map or Set would be better here. Due to direct access to this list one is able to build inconsistent multiple options of the same type using getOptions or setOptions.

File: pom.xml (hunk 1 above)
Comment (AnEmortalKid) [reply pom.xml:416]: Ah, when i initially try to do something with the parser I like to have the DEBUG_PARSER and DEBUG_LOOKAHEAD options set to true. When you run `mvn clean package` with those, the build takes quite a while (mostly from all the printing). I can revert these since they're only just for my own benefit when trying to add new things, and it's mostly cause i set all the debug options!

File: src/main/java/net/sf/jsqlparser/statement/ExplainStatement.java (hunk 2 above)
Comment (AnEmortalKid) [reply src/main/java/net/sf/jsqlparser/statement/ExplainStatement.java:23]: Umm yeah, I think you're right, it can only be a LinkedSet (order preserved). I'll change that up!

File: pom.xml (hunk 1 above)
Comment (wumpz) [reply pom.xml:417]: No if you find them useful keep them, but keep the standard behaviour untouched.

Comment (AnEmortalKid) [reply pom.xml:417]: cool, updated!

File: src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt [hunk 3]
```diff
@@ -635,13 +643,109 @@ DescribeStatement Describe(): {
...
+
+/**
+ * The output format, which can be TEXT, XML, JSON, or YAML
+ */
+String ExplainFormatOption():
+{
+  Token tk = null;
+}
+{
+   // TODO support Text
```
Comment (AnEmortalKid) [inline src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:685]: ran into issues with conflicting "TEXT", so i've just left it out atm until someone requests it.
